pkg/cmd/cli/cmd/logs.go

```diff
@@ -63,21 +63,26 @@ type OpenShiftLogsOptions struct {
 }
 
 // NewCmdLogs creates a new logs command that supports OpenShift resources.
-func NewCmdLogs(name, parent string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+func NewCmdLogs(name, baseName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
 	o := OpenShiftLogsOptions{
 		KubeLogOptions: &kcmd.LogsOptions{},
 	}
+
 	cmd := kcmd.NewCmdLogs(f.Factory, out)
 	cmd.Short = "Print the logs for a resource."
 	cmd.Long = logsLong
-	cmd.Example = fmt.Sprintf(logsExample, parent+" "+name)
+	cmd.Example = fmt.Sprintf(logsExample, baseName, name)
 	cmd.SuggestFor = []string{"builds", "deployments"}
 	cmd.Run = func(cmd *cobra.Command, args []string) {
-		kcmdutil.CheckErr(o.Complete(f, out, cmd, args))
+		if err := o.Complete(f, cmd, args, out); err != nil {
+			kcmdutil.CheckErr(err)
+		}
 		if err := o.Validate(); err != nil {
 			kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
 		}
-		kcmdutil.CheckErr(o.RunLog())
+		if err := o.RunLog(); err != nil {
```

Comment:
We've actually settled back in to using kcmdutil.CheckErr() even on nil output - no if block necessary.